Simplified/Reader2/LCP/LCPPassphraseAuthenticationService.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  LCPPassphraseAuthenticationService.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2021-02-08.
+//  Copyright © 2021 NYPL Labs. All rights reserved.
+//
+
+#if LCP
+
+import Foundation
+import ReadiumLCP
+
+class LCPPassphraseAuthenticationService: LCPAuthenticating {
+  func retrievePassphrase(for license: LCPAuthenticatedLicense, reason: LCPAuthenticationReason, allowUserInteraction: Bool, sender: Any?, completion: @escaping (String?) -> Void) {
+    guard let hintLink = license.hintLink,
+      let hintURL = URL(string: hintLink.href) else {
+      Log.error(#file, "LCP Authenticated License does not contain valid hint link")
+      completion(nil)
+      return
+    }
+    
+    NYPLNetworkExecutor.shared.GET(hintURL) { (result) in
+      switch result {
+      case .success(let data, _):
+        if let json = try? JSONSerialization.jsonObject(with: data, options: []) as! [String:Any],
+          let passphrase = json["passphrase"] as? String {
+          completion(passphrase)
+        } else {
+          Log.error(#file, "Error parsing JSON or finding passphrase key/value.")
```

Comment:
should we log an event instead, since it seems like a rather serious error? That way we could log the whole json maybe (unless it's really huge and has sensitive info).

Reply:
The `json` won't be available since it's outside of the `if-let`, should I log the plain `data`? But it won't be readable on Crashlytics...

Reply:
doh you're right... you could log the data converted to string though.